clap_generate/src/man.rs

```diff
@@ -0,0 +1,287 @@
+            None => buf.push_str(r#" """#),
+        }
+
+        match self.manual.as_ref() {
+            Some(manual) => write(buf, format_args!(r#" "{}""#, manual)).unwrap(),
+            None => buf.push_str(r#" """#),
+        }
+
+        buf.push('\n');
+    }
+
+    fn render_about(&self, buf: &mut String) {
+        buf.push_str(".SH \"NAME\"\n");
+        buf.push_str(&bold(self.app.get_name()));
+
+        if let Some(about) = self.app.get_about() {
+            buf.push_str(r" \- ");
+            buf.push_str(&escape(about));
+        }
+
+        buf.push('\n');
+    }
+
+    fn render_description(&self, buf: &mut String) {
+        if let Some(about) = self.app.get_long_about() {
+            buf.push_str(".SH \"DESCRIPTION\"\n");
+            buf.push_str(about);
+            buf.push('\n');
+        }
+    }
+
+    fn render_synopsis(&self, buf: &mut String) {
+        buf.push_str(".SH \"SYNOPSIS\"\n");
+        buf.push_str(&italic(self.app.get_name()));
+        buf.push(' ');
+
+        for opt in self.app.get_arguments() {
+            match (opt.get_short(), opt.get_long()) {
+                (Some(short), Some(long)) => {
+                    write(buf, format_args!("[-{}|--{}] ", short, long)).unwrap()
+                }
+                (Some(short), None) => write(buf, format_args!("[-{}] ", short)).unwrap(),
+                (None, Some(long)) => write(buf, format_args!("[--{}] ", long)).unwrap(),
+                (None, None) => (),
+            }
+        }
+
+        for arg in self.app.get_positionals() {
+            write(buf, format_args!("[{}]", arg.get_name())).unwrap();
+        }
+
+        buf.push('\n');
+    }
+
+    // TODO: handle positional arguments in a different way?
+    fn render_options(&self, buf: &mut String) {
+        let items: Vec<_> = self
+            .app
+            .get_arguments()
+            .filter(|i| !i.is_set(ArgSettings::Hidden))
+            .collect();
+        if items.is_empty() {
+            return;
+        }
+
+        buf.push_str(".SH \"OPTIONS\"\n");
+        for opt in items.iter().filter(|a| !a.is_positional()) {
+            let args = match (opt.get_short(), opt.get_long()) {
+                (Some(short), Some(long)) => format!("-{}, --{}", short, long),
+                (Some(short), None) => format!("-{}", short),
+                (None, Some(long)) => format!("--{}", long),
+                (None, None) => "".to_string(),
+            };
+
+            buf.push_str(&bold(&args));
+
+            if let Some(value) = &opt.get_value_names() {
+                buf.push(' ');
+                buf.push_str(&italic(&value.join(" ")));
+            }
+
+            buf.push('\n');
+
+            if let Some(help) = opt.get_help() {
+                buf.push_str(&indent(4, help));
+            }
+
+            if let Some(help) = opt.get_long_help() {
+                buf.push_str("\n.sp\n.sp\n");
+                buf.push_str(&indent(4, help));
+            }
```

Comment:
When we generate the help, we also include
- default values
- relevant env variable (granted, we shouldn't include the current env value)
- possible values
  - A question for possible values is they *can* have help text.  This is currently only being used by the completion scripts.  Do assume that the `long_help` will include that information or should we detect when help for possible values is included and show the possible values as a list with the associated help text

Should we be including these?

Reply:
I agree that including default values and environment variables are useful, not so sure about how including possible values would look though. I'll give it a go!

Reply:
Also, there are hidden settings for all of those, and some also have a global version (`AppSettings::HidePossibleValues`).

Reply:
The current implementation shows the default values in a `[default: <values>]` block after the option and has a paragraph added to the bottom about the environment variable if it exists and is visible.

Reply:
What about possible values, their help message, and the various hide settings for envs, defaults, values, etc.

Oh, and should visible aliases be shown?